sbom-utility-scripts/scripts/base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -127,10 +168,96 @@ def main():
         sbom = json.load(f)
 
     base_images_sbom_components = get_base_images_sbom_components(base_images_digests, is_last_from_scratch)
-    if "formulation" in sbom:
-        sbom["formulation"].append({"components": base_images_sbom_components})
+    if detect_sbom_format(sbom) == "cyclonedx":
+        if "formulation" in sbom:
+            sbom["formulation"].append({"components": base_images_sbom_components})
+        else:
+            sbom.update({"formulation": [{"components": base_images_sbom_components}]})
     else:
-        sbom.update({"formulation": [{"components": base_images_sbom_components}]})
+        root_element1, map1, inverse_map1 = map_relationships(sbom["relationships"])
+
+        packages = []
+        relationships = []
+
+        # Try to calculate root package represeting the container image or directory, which was
+        # used to build the SBOM, based on the relationships maps.
+        # SPDX has relationsship ROOT-ID DESCRIBES MIDDLE-ID which express the fact the SBOM documents
+        # describes container image or directory represented by MIDDLE-ID package.
+        root_package1 = None
+        for r, contains in map1.items():
+            # root package is the one which contains another elements and is in relationship with
+            # the document element where it stand as relatedSpdxElement
+            if contains and inverse_map1.get(r) == root_element1:
+                root_package1 = r
```

Comment:
> SPDX has relationsship ROOT-ID DESCRIBES MIDDLE-ID

The goal is straightforward, but the code is not. Could you simplify the code to just find the `ROOT-ID DESCRIBES MIDDLE-ID` relationship?

Reply:
In [add_image_reference.py](https://github.com/konflux-ci/build-tasks-dockerfiles/blob/main/sbom-utility-scripts/scripts/add-image-reference-script/add_image_reference.py), we also have handling for the case when there are multiple `ROOT-ID DESCRIBES *` relationships. Should we:

* raise an error here if that happens, always call this script _after_ `add_image_reference.py`
* have similar handling here as `add_image_reference.py`
* other options?

Reply:
I think it makes sense to raise exception in such case as many other scripts depend on the same condition